### PR TITLE
Add ip route append

### PR DESCRIPTION
**Pull request by yaocw2020**
Add ip route append, equivalent to: `ip route append $route`

Signed-off-by: yaocw2020 <yaocanwu@gmail.com>

### CODE REVIEW

**Comment by yaocw2020** [comment]
https://github.com/vishvananda/netlink/issues/539

**Comment by yaocw2020** [comment]
Go test result
```
$ go test -v . -run TestRouteAppend
=== RUN   TestRouteAppend
--- PASS: TestRouteAppend (0.00s)
PASS
ok  	github.com/vishvananda/netlink	(cached)
```

**Comment by yaocw2020** [comment]
@aboch We use `netlink` to implement [Harvester](https://github.com/rancher/harvester) network controller. We found that `RouteAdd` and `RouteReplace` can not meet the requirements, while `ip route append $route` can. Please review this PR, thanks!

File: route_test.go [hunk 1]
```diff
@@ -206,6 +206,70 @@ func TestRouteReplace(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	routes, err = RouteList(link, FAMILY_V4)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(routes) != 2 || !routes[1].Src.Equal(ip) {
+		t.Fatal("Route not replaced properly")
```
Comment (aboch) [inline route_test.go:255]: Just a not, this was not a replace operation

**Review by aboch — APPROVED** [review]

**Comment by aboch** [comment]
LGTM

**Pull request merged**